### ISSUE TITLE
Xiaomi AC2100 missing internet status information

### ISSUE THREAD

**Issue by minax007** [hanwckf/rt-n56u#530]
Dear @hanwckf, Dear all,

my Xiami Mi AC2100 (R2100) and my Xiaomi Mi 3C are running Padavan from hanwckf. 
I have noticed a difference in the "Internet Status" information. 
The Xiaomi Mi 3C shows in addition to the R2100 following information:
**- Traffic During The Session
- Current Data Rate**

See red marked entries: 

![Screenshot_2020-12-29 MI-3C Wireless Router](https://user-images.githubusercontent.com/67478561/103300204-a7c29480-49fe-11eb-9da8-2bb2603fbf05.png)

I would really love to have this information also for the R2100. 

Does anybody know if this is possible and how I can get this to be displayed?

Thanks in advance!

**Comment by CessXP** [comment]
Hi, I want to question something for you, do you have problems with 5Ghz Wifi?

**Comment by minax007** [comment]
Dear @CessXP 
I have no problems with it, because I am not using it. It is deactivated. I have activated it to see if that was the reason for the missing information - but it hasn't changed.
Please let me know why you are asking. 
Thanks!

**Comment by minax007** [comment]
By the way. The screenshot shown above is from the MI 3C router, which does not have 5 GHz. 
On the R2100 (which has 5 GHz) the information shown in the red box in the screenshot do not get displayed.

**Comment by CessXP** [comment]
> Dear @CessXP
> I have no problems with it, because I am not using it. It is deactivated. I have activated it to see if that was the reason for the missing information - but it hasn't changed.
> Please let me know why you are asking.
> Thanks!

Oh, I'm using 5ghz WiFi and if I move some meters from the router, I have a lot of packages lost until gets stabilize but if I move again the problem back.

**Comment by minax007** [comment]
Seems to be a driver issue. Also OpenWRT does not perform as good as the stock firmware in the 5 GHz band. 
I prefer 2,4 GHz, because I do not need high bandwidth. 
And Padavan from hanwckf works perfectly fine in the 2,4 GHz band. 

By the way, do you know how I can get the following information displayed for the R2100:
**- Traffic During The Session**
**- Current Data Rate**

**Comment by CessXP** [comment]
Yes, actually in network traffic menu, there are graphics from every port and WiFi, even shows actual traffic and some other info, maybe works for you.

**Comment by CessXP** [comment]
> Seems to be a driver issue. Also OpenWRT does not perform as good as the stock firmware in the 5 GHz band.
> I prefer 2,4 GHz, because I do not need high bandwidth.
> And Padavan from hanwckf works perfectly fine in the 2,4 GHz band.
> 
> By the way, do you know how I can get the following information displayed for the R2100:
> **- Traffic During The Session**
> **- Current Data Rate**
I hope the driver get fixed, actually Im in Openwrt with the yesterday snapshot and its ok, no the best but ok

**Comment by minax007** [comment]
Dear @CessXP 
thanks for the information. But that was not what I was looking for. The "Network Traffic" overview works fine on the firmware. I am looking for the information shown on the picture above in the "Network Map" "Internet Status" overview.

**Comment by CessXP** [comment]
> Dear @CessXP
> thanks for the information. But that was not what I was looking for. The "Network Traffic" overview works fine on the firmware. I am looking for the information shown on the picture above in the "Network Map" "Internet Status" overview.

Damn maybe it is disabled in the firmware, you can compare the config files of both and check what it could be

**Comment by minax007** [comment]
Dear @CessXP 
I have compared config files already, but couldn't figure out what it is. 
The entries are being created in the following ASP-file: http://my.router/device-map/internet.asp
But I did not find out what triggers this ASP-file to display those entries.

**Comment by zouyonghao** [comment]
Actually I do not see this information with your 3C firmware, weird.

**Comment by minax007** [comment]
Dear @zouyonghao 
Thanks for having a look on the issue I had posted!
I am using the 3C router in "Repeater-Mode" (AP-Bridge & Client). 
Maybe that is the reason. 
Nevertheless I like to see with figures (not only graphs as shown under "Network Traffic").

**Comment by zouyonghao** [comment]
> Dear @zouyonghao
> Thanks for having a look on the issue I had posted!
> I am using the 3C router in "Repeater-Mode" (AP-Bridge & Client).
> Maybe that is the reason.
> Nevertheless I like to see with figures (not only graphs as shown under "Network Traffic").

I think the logic here is in [web_ex.c](https://github.com/hanwckf/rt-n56u/blob/master/trunk/user/httpd/web_ex.c)

The variable `wan_bytes_rx` and `wan_bytes_tx` contain this information, and they are displayed if these two are greater than 0.

**Comment by minax007** [comment]
Dear @zouyonghao 
Thanks for caring and looking into the details!
I had figured out those variables and I see that those need to be greater than 0. 
But with the same code those lines to get displayed for the Mi-3C and not for R2100. 
This is what I couldn't figure out yet. 
Both routers are running with the same configuration "WAN (Wireless-ISP)" in repeater mode.

**Comment by CessXP** [comment]
I noticed R2100 dont show the info in Ap mode or repeater mode

**Comment by minax007** [comment]
Dear @CessXP 
exactly, this is also what I have found out. 
But I couldn't figure out why ... 
I guess this problem also occurs with the RM2100 (white AC2100 with six antennas), because it is very similar.

**Comment by CessXP** [comment]
> Dear @CessXP
> exactly, this is also what I have found out.
> But I couldn't figure out why ...
> I guess this problem also occurs with the RM2100 (white AC2100 with six antennas), because it is very similar.

Actually yes, the same thing happens

**Comment by minax007** [comment]
Dear @yuos-bit
today I have tested your Padavan firmware for the Mi-3C and see that on the "Internet Status" page the following information is also not displayed (picture is in the first post of this thread):
- Traffic During The Session
- Current Data Rate

I know that you had forked the repository from hiboyhiboy, which is an older fork from the hanwckf repository.  

Afterwards I have tested the Mi-3C with latest repository from hanwckf and see that the information mentioned before is being displayed. 

Do you have a clue what the reason could be why it is not displayed in your firmware ... or what is necessary to have it displayed?

The reason why I am asking is that I have another router (R2100) where this information does not get displayed even with the hanwckf firmware ... and I can't figure out why :-)

**Comment by yuos-bit** [comment]
Dear friends, you can go to my GitHub RT-N56U project to download my MI-3C firmware and try it out, but before that, you need to restore the factory Settings to the public version under Breed.



---Original---

**Comment by minax007** [comment]
Dear @yuos-bit 
this is what I did yesterday. And I see that the information does not get displayed on my Mi-3C. 
Before installing the firmware I had made a reset in BREED.

**Comment by yuos-bit** [comment]
As you know, you can go to my GitHub and then download my compiled image from his post.



---Original---

**Comment by minax007** [comment]
Dear @yuos-bit 
I know my dear friend. 
Do you get the information displayed on your Mi-3C with your firmware?

**Comment by yuos-bit** [comment]
Yes, my friend. At least in my display



---Original---

**Comment by minax007** [comment]
Dear @yuos-bit 
did you recently do changes with regards to displaying this information?
I see that with the firmware of yours from September 16th, and also former versions of yours, those information do not get displayed for the Mi-3C.

**Comment by yuos-bit** [comment]
please look here,this is my github releases.


https://github.com/yuos-bit/rt-n56u/releases/tag/v3.4.3-099







---Original---

**Comment by zouyonghao** [comment]
I get this information after reading the code carefully.

Try login in to your router via `ssh` and use `nvram set hw_nat_mode=2`, then this info shows.

**Comment by minax007** [comment]
Dear @zouyonghao 

thanks for your response!

I have started ssh and have entered `nvram show` to check the actual state. 

The result shows already `hw_nat_mode=2`.

**Comment by minax007** [comment]
Dear @zouyonghao 
the settings in http://my.router/Advanced_WAN_Content.asp already provide the option to set the `hw_nat_mode`. 

![grafik](https://user-images.githubusercontent.com/67478561/104308948-b23a5f00-54d1-11eb-9288-b765fcddc1c3.png)

If the option shown in the picture is set to "Disable (Slow)" then you the `hw_nat_mode` is set to `2`.

**Comment by minax007** [comment]
This is the code for this parameter:
```
<select name="hw_nat_mode" class="input">
  <option value="0" <% nvram_match_x("", "hw_nat_mode", "0", "selected"); %>>Offload TCP for LAN</option>
  <option value="1" <% nvram_match_x("", "hw_nat_mode", "1", "selected"); %>>Offload TCP for LAN/WLAN</option>
  <option value="3" <% nvram_match_x("", "hw_nat_mode", "3", "selected"); %>>Offload TCP/UDP for LAN</option>
  <option value="4" <% nvram_match_x("", "hw_nat_mode", "4", "selected"); %>>Offload TCP/UDP for LAN/WLAN</option>
  <option value="2" <% nvram_match_x("", "hw_nat_mode", "2", "selected"); %>>Disable (Slow)</option>
</select>
```

**Comment by zouyonghao** [comment]
> Dear @zouyonghao
> the settings in http://my.router/Advanced_WAN_Content.asp already provide the option to set the `hw_nat_mode`.
> 
> ![grafik](https://user-images.githubusercontent.com/67478561/104308948-b23a5f00-54d1-11eb-9288-b765fcddc1c3.png)
> 
> If the option shown in the picture is set to "Disable (Slow)" then you the `hw_nat_mode` is set to `2`.

This is not shown on my router...
But I got the traffic information after I set the nvram parameter.

If you review the code, it seems these variables are not calculated if it is ap_mode.

**Comment by minax007** [comment]
Dear @zouyonghao 
which operation mode is your router set to?
My router is used in "AP-Client + AP" (Repeater) mode. 
In my case it doesn't make a difference which `hw_nat_mode` I set.

**Comment by zouyonghao** [comment]
> Dear @zouyonghao
> which operation mode is your router set to?
> My router is used in "AP-Client + AP" (Repeater) mode.
> In my case it doesn't make a difference which `hw_nat_mode` I set.

Wireless Router Mode (Default)

**Comment by minax007** [comment]
Does the information get displayed in the Wireless Router Mode when `hw_nat_mode` is set to `2`?

**Comment by minax007** [comment]
And are you using the R2100 or another router?
Thanks in advance!

**Comment by zouyonghao** [comment]
> Does the information get displayed in the Wireless Router Mode when `hw_nat_mode` is set to `2`?

Yes for me. I'm using Mi-3C...

**Comment by minax007** [comment]
I agree with my Mi-3C the information also does get displayed. But with the R2100 (Mi AC2100) it does not. Even though I am using the same system settings. This is very strange :-)

**Comment by zouyonghao** [comment]
And I try to set my RM2100 to Wireless Router Mode and set `hw_nat_mode` to `2`, the information appeared.

**Comment by minax007** [comment]
Dear @zouyonghao ,
thanks for checking and finding a solution for the RM2100!

The RM2100 is from architectural point of view almost identical to the R2100. The parameterization is therefore identical. 
So I am wondering why it works on your device and not on mine.

**Comment by minax007** [comment]
Today I have tried with three differend Padavan builds (also builds, which I did not compile on my own) in different operation modes. 
I have set `nvram set hw_nat_mode=2` via user interface, and also via ssh. 
But the information does not get displayed.
